### PR TITLE
Feature/#668 dn is up to date

### PR DESCRIPTION
**Pull request by toan-quach**
#668

### CODE REVIEW

File: src/taipy/core/data/data_node.py [hunk 1]
```diff
@@ -567,6 +568,22 @@ def is_up_to_date(self) -> bool:
             return False
         return True
 
+    @property
+    def is_up_to_date(self) -> bool:
```
Comment (trgiangdo) [inline src/taipy/core/data/data_node.py:572]: Please add a docstring for the refman to explain this property

File: src/taipy/core/data/data_node.py [hunk 2]
```diff
@@ -549,7 +550,7 @@ def is_ready_for_reading(self) -> bool:
 
     @property  # type: ignore
     @_self_reload(_MANAGER_NAME)
-    def is_up_to_date(self) -> bool:
+    def is_valid(self) -> bool:
         """Indicate if this data node is up-to-date.
```
Comment (trgiangdo) [inline src/taipy/core/data/data_node.py]: The docstring here can be misleading since we have a new `is_up_to_date` property

**Review by trgiangdo — COMMENTED** [review]
I have a few comments about the refman

**Review by trgiangdo — APPROVED** [review]
Both the document and the code looks good now. Nice work

**Review by jrobinAV — APPROVED** [review]
Neat!

**Pull request merged**